### PR TITLE
feat: support sentry reporting

### PR DESCRIPTION
**Pull request by asbiin**
You fool, don't forget these steps:

- [x] Unit tests
- [ ] Tests with Cypress
- [ ] Documentation
- [ ] Dummy data

### CODE REVIEW

File: app/Http/Middleware/SentryContext.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    public function handle($request, Closure $next)
+    {
+        if (app()->bound('sentry') && config('app.sentry.enable')) {
+            \Sentry\configureScope(function (Scope $scope): void {
+                // Add user context
+                $user = Auth::user();
+                if ($user !== null) {
+                    $scope->setUser([
+                        'id' => $user->id,
+                        'email' => $user->email,
```
Comment (djaiss) [inline app/Http/Middleware/SentryContext.php:31]: I don’t think we should do that, nor storing the name.
If we do that, it’ll be really tough to be GDPR compliant. Also, as we'll be in different countries, law differ per country.
We should only log anonymous data in an external system that we don’t control. That means, I would only store the user id.

**Review by djaiss — CHANGES_REQUESTED** [review]
Awesome. I've added one super important comment.
Also, we should update the .env.example as well with default values --- shouldn't we?

File: app/Http/Middleware/SentryContext.php (hunk 1 above)
Comment (asbiin) [reply app/Http/Middleware/SentryContext.php:31]: We could log only the id if you prefer, but I'm not sure GDPR has something against that.

Comment (djaiss) [reply app/Http/Middleware/SentryContext.php:31]: You are right for GDPR specifically for this use case. But, in Canada for example, most businesses refuse to host data about someone in a system where the data is not located in Canada. If we were to send name and email in a system where the DB is in the US for example, we won’t be able to do business with it as they'll refuse.
So we need to make sure that every personal data about someone stays in a system that we control.

Comment (asbiin) [reply app/Http/Middleware/SentryContext.php:31]: OK so in that case we could left the `id` at least

**Review by djaiss — APPROVED** [review]
👍

**Pull request merged**